### ISSUE TITLE
Service unavailable for legal reasons

### ISSUE THREAD

**Issue by Alexey71** [tailscale/tailscale#13648]
### What is the issue?

I cant connect.
"Service unavailable for legal reasons"
Doesn't work in Russia anymore?

**Comment by tyrylgin** [comment]
Yep, control plane not accessible from territory of Russia anymore. So, you should host Headscale now 
https://www.reddit.com/r/Tailscale/comments/1fu0iwk/seems_tailscale_geoblocked_russia_completely/

**Comment by mdubakin** [comment]
We have our own Headscale deployed, but Tailscale clients are also unable to connect to it.

Here is an example of the Headscale logs:

```
Oct 02 12:09:15 headscale-1 headscale[698]: 2024-10-02T12:09:15Z INF The client has closed the connection handler=PollNetMapStream machine=MacBook-Pro-3
Oct 02 12:13:23 headscale-1 headscale[698]: 2024-10-02T12:13:23Z INF ../../../home/runner/work/headscale/headscale/hscontrol/protocol_common.go:703 > Machine successfully authorized machine=MacBook-Pro-3 noise=true
Oct 02 12:13:23 headscale-1 headscale[698]: 2024-10-02T12:13:23Z INF Client sent endpoint update and is ok with a response without peer list handler=PollNetMap machine=MacBook-Pro-3 noise=true
Oct 02 12:13:24 headscale-1 headscale[698]: 2024-10-02T12:13:24Z INF Client is ready to access the tailnet handler=PollNetMap machine=MacBook-Pro-3 noise=true
Oct 02 12:13:24 headscale-1 headscale[698]: 2024-10-02T12:13:24Z INF Sending initial map handler=PollNetMap machine=MacBook-Pro-3 noise=true
Oct 02 12:13:24 headscale-1 headscale[698]: 2024-10-02T12:13:24Z INF Notifying peers handler=PollNetMap machine=MacBook-Pro-3 noise=true
```

However, Tailscale cannot connect:

```
$ tailscale status
100.64.0.10     macbook-pro-3        s18174732    macOS   -
100.64.0.1      gw-0.admin.sysadmin.local admin        linux   -

# Health check:
#     - no DERP home

$ tailscale netcheck
2024/10/02 16:18:38 No DERP map from tailscaled; using default.
2024/10/02 16:18:40 fetch prodDERPMap: 451 Unavailable For Legal Reasons: Service unavailable for legal reasons

$ tailscale ping 100.64.0.1
ping "100.64.0.1" timed out
ping "100.64.0.1" timed out
ping "100.64.0.1" timed out
```

I would like to note that the DERP is disabled on the Headscale server:

```
...
derp:
  server:
    enabled: False
...
```

Could you please let me know if your changes might have affected the self-hosted headscale?

**Comment by and7ey** [comment]
There should be an official announcement about such changes.

**Comment by he0s** [comment]
And one more service decided to help the dictatorship regime to limit freedom of speech and access to alternative sources of information. It's a sign you have FSB agents in your headquarter. :)

**Comment by detectivecider** [comment]
Is it only russia? I can't access it within Iraq

**Comment by he0s** [comment]
@detectivecider well, I can't speak for other countries, but it seems like access limited in the same way.

**Comment by detectivecider** [comment]
It seems everything works fine now, but it would be nice to know if this was for anything actually related to laws or something broke and this was a fallback text?

**Comment by todayer** [comment]
The official response in the mail:

Hi There, 

 

As part of our ongoing legal compliance efforts, we’re taking a closer look at our legal requirements as a company operating around the globe. A code update as part of that work went out earlier than expected, and as a result, users in certain jurisdictions were blocked from accessing Tailscale. 

 

We apologize for not providing advanced notice of this change and the resulting impact this may have caused. We’re working to improve our proactive communications to provide a better customer experience in the future.

 

To minimize the disruption, we’ve rolled back the code update for the time being. As a legal obligation, we’ll still need to implement these changes, but we’ll do so at a future date. When that happens, we’ll provide notification ahead of time and be available to help.


Sincerely,

Tailscale

**Comment by lifeart** [comment]
Wondering, how this `commitment` corresponds to https://www.reuters.com/technology/us-calls-big-tech-help-evade-online-censors-russia-iran-2024-09-05/ initiative.

I do understand that paid services should not be provided (make sense), but why not leave free tier for Russian users?

**Comment by lifeart** [comment]
Wondering, how this `commitment` corresponds to https://www.reuters.com/technology/us-calls-big-tech-help-evade-online-censors-russia-iran-2024-09-05/ initiative.

I do understand that paid services should not be provided (make sense), but why not leave free tier for Russian users?

**Comment by sevenrats** [comment]
Tailscale is a Canadian company. Presumably they are not allowed to do business in countries that have active trade sanctions in place.

**Comment by detectivecider** [comment]
In the case of sanctions in russia, google playstore has let users download free content and content that they have already paid for, but disallows them from making new purchases. Apple seems to outright don't care about the sanctions when it comes to their appstore.

Of course apple complies with the russian government because you know...it is apple.

but i don't see why tailscale couldn't implement what google does with the playstore.

**Comment by kytta** [comment]
> Tailscale is a Canadian company. Presumably they are not allowed to do business in countries that have active trade sanctions in place.

I guess they should start blocking USA soon ;)

---

Anyway, if you have access to a machine outside Russia, you can download the packages and copy them over to your server. For example, I am running a Debian 12 server in Russia, and I have access to a Debian 12 machine in Germany. On a German machine, I run:

```bash
# https://tailscale.com/kb/1174/install-debian-bookworm
curl -fsSL https://pkgs.tailscale.com/stable/debian/bookworm.noarmor.gpg | sudo tee /usr/share/keyrings/tailscale-archive-keyring.gpg >/dev/null
curl -fsSL https://pkgs.tailscale.com/stable/debian/bookworm.tailscale-keyring.list | sudo tee /etc/apt/sources.list.d/tailscale.list

sudo apt-get install --print-uris tailscale
```

This will not install Tailscale, but print the URIs to the `.deb` packages, as well as other, normal Debian, dependencies. Download the ones that are from Tailscale:

```bash
wget https://pkgs.tailscale.com/stable/debian/pool/tailscale_1.80.3_amd64.deb
wget https://pkgs.tailscale.com/stable/debian/pool/tailscale-archive-keyring_1.35.181_all.deb
```

And then copy them to your Russian server:

```bash
scp tailscale_1.80.3_amd64.deb tailscale-archive-keyring_1.35.181_all.deb <IP of your server>:<optional path on the server>
```

Ok, now let's switch to the server in Russia. First, install the dependencies using your functioning Debian mirrors:

```bash
sudo apt install iptables libip6tc2 libjansson4 libnetfilter-conntrack3 libnfnetlink0 libnftables1 libnftnl11 nftables
```

Then, install the copied `.deb`s with `dpkg`:, and start Tailscale

```bash
sudo dpkg -i tailscale_1.80.3_amd64.deb tailscale-archive-keyring_1.35.181_all.deb

sudo tailscale up
```

Et voilà, your Russian server now runs legit Tailscale. These steps should work for other Debian-based distros, like Ubuntu, but keep in mind that the URIs for the `.deb`s might be different. Similar steps should work on RPM-based package managers, too, but I don't know how.

**Q: Why not just the static binaries?**\
**A:** I don't want to set up systemd etc. myself. The `.deb` already ships with everything I need

**Q: Will this auto-update?**\
**A:** No, you'll have to do it all over again

**Comment by he0s** [comment]
@kytta, I guess your original family name is Morozov. :)

**Comment by kytta** [comment]
Huh, I wonder why my comment was deleted
